Additions/XCTestCase-KIFAdditions.m

```diff
@@ -34,7 +34,18 @@ - (void)failWithException:(NSException *)exception stopTest:(BOOL)stop
 {
     self.continueAfterFailure = YES;
 
-    [self recordFailureWithDescription:exception.description inFile:exception.userInfo[@"FilenameKey"] atLine:[exception.userInfo[@"LineNumberKey"] unsignedIntegerValue] expected:NO];
+    NSString *filePath = exception.userInfo[@"FilenameKey"];
+    NSInteger lineNumber = [exception.userInfo[@"LineNumberKey"] unsignedIntegerValue];
+    XCTSourceCodeLocation *location = [[XCTSourceCodeLocation alloc] initWithFilePath:filePath
+                                                                           lineNumber:lineNumber];
+    XCTSourceCodeContext *context = [[XCTSourceCodeContext alloc] initWithLocation:location];
+    XCTIssue *issue = [[XCTIssue alloc] initWithType:XCTIssueTypeAssertionFailure
+                                  compactDescription:exception.description
+                                 detailedDescription:nil
+                                   sourceCodeContext:context
+                                     associatedError:nil
+                                         attachments:@[]];
```

Comment:
Out of scope - It would potentially make sense to attach the generated test failure screenshot and the view hierarchy as attachments at some point. Not in this PR though.

Reply:
That would be fancy!